riptide-nakadi/src/main/java/org/zalando/riptide/nakadi/NakadiGateway.java

```diff
@@ -0,0 +1,179 @@
+package org.zalando.riptide.nakadi;
+
+/*
+ * ⁣​
+ * Riptide: Nakadi
+ * ⁣⁣
+ * Copyright (C) 2015 - 2016 Zalando SE
+ * ⁣⁣
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ​⁣
+ */
+
+import static org.springframework.http.HttpStatus.Series.SUCCESSFUL;
+import static org.springframework.http.MediaType.parseMediaType;
+import static org.zalando.riptide.Bindings.anySeries;
+import static org.zalando.riptide.Bindings.on;
+import static org.zalando.riptide.Navigators.contentType;
+import static org.zalando.riptide.Navigators.series;
+import static org.zalando.riptide.Route.responseEntityOf;
+import static org.zalando.riptide.RoutingTree.dispatch;
+import static org.zalando.riptide.stream.Streams.streamOf;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus.Series;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.client.ClientHttpResponse;
+import org.springframework.stereotype.Component;
+import org.zalando.problem.ThrowableProblem;
+import org.zalando.riptide.Binding;
+import org.zalando.riptide.Bindings;
+import org.zalando.riptide.Rest;
+import org.zalando.riptide.ThrowingConsumer;
+import org.zalando.riptide.capture.Capture;
+
+@Component
+public class NakadiGateway {
+
+    public static final MediaType PROBLEM = parseMediaType("application/problem+json");
+
+    private static final Logger LOG = LoggerFactory.getLogger(NakadiGateway.class);
+    private static final String SESSION_HEADER = "X-Nakadi-SessionId"; // TODO should this be X-Nakadi-StreamId?
+
+    private static final Binding<Series> ON_FAILURE_BINDING =
+            anySeries().call(dispatch(contentType(),
+                    on(PROBLEM).call(ThrowableProblem.class, propagate()),
+                    Bindings.anyContentType().call(invalidContent())));
+
+    private static ThrowingConsumer<ClientHttpResponse> invalidContent() {
+        return response -> {
+            InputStreamReader input = new InputStreamReader(response.getBody());
```

Comment:
Although an option. Better is @whiskeysierra suggested:
```
new String(ByteStreams.toByteArray(response.getBody()), charset)
```
I like the above option, because it is without guava :)